test/cpp/tensorexpr/test_ops.cpp

```diff
@@ -41,3 +41,49 @@ TEST(Ops, Sum) {
     ASSERT_TRUE(at::allclose(bt, ref));
   }
 }
+
+TEST(Ops, Stack) {
+  constexpr int N = 2;
+  std::vector<IntList> testDims = {{0}, {1}, {0, 1}, {1, 0}, {2, 3}, {3, 2, 2}};
+  std::vector<std::vector<ExprHandle>> outputShapes = {
+      {N, ExprHandle(0)},
+      {N, ExprHandle(1)},
+      {N, ExprHandle(0), ExprHandle(1)}};
+  for (int idx = 0; idx < testDims.size(); idx++) {
+    // Construct the input buffer list
+    const auto& dims = testDims[idx];
+    std::vector<ExprHandle> inShape;
+    for (auto d : dims) {
+      inShape.push_back(ExprHandle(d));
+    }
+
+    BufHandle a("a", inShape, kFloat);
+    BufHandle b("b", inShape, kFloat);
+    std::vector<BufHandle> buflist;
+    buflist.push_back(a);
+    buflist.push_back(b);
+
+    auto at = at::rand(dims, at::kFloat);
+    auto bt = at::rand(dims, at::kFloat);
```

Comment:
The test only checks stacking two tensors with equal dimensions. We should also check other cases, e.g.:
1) stacking just one tensor
2) stacking N>2 tensors
3) stacking tensors with different dimensions

It would be good to also have such tests in python, so that whenever opinfo covers `stack` we can use that too.

It might be also a good idea to have a test verifying IR that we generate.

Reply:
> The test only checks stacking two tensors with equal dimensions. We should also check other cases...

That's a good point! Will add more tests.

> It would be good to also have such tests in python, so that whenever opinfo covers `stack` we can use that too.

Not sure if I understand this. Are you suggesting to test above cases from Python instead of C++? We don't have many unit tests for ops right now. I only know aten::sum, aten::cat, and aten::conv. For these 3 ops, we put the unit tests in 3 files under `test/cpp/tensorexpr`. Maybe we should put them together in one file like `test_ops.cpp`. 

> It might be also a good idea to have a test verifying IR that we generate.

Agree. Will add it.

Reply:
> Are you suggesting to test above cases from Python instead of C++? 

Having C++ tests is fine, but currently the most comprehensive coverage that we have is in python:
https://github.com/pytorch/pytorch/blob/122f8648ab8f872bd8ed970728c5ac894dce5300/test/test_jit_fuser_te.py#L2476

It leverages OpInfo, which is basically a collection of representative inputs for an op. What I suggested was to add a test for `stack` to those tests as well (if possible). If `stack` is not supported by OpInfo, then I think it still would make sense to write a python test manually, as most of our ops are currently tested from python rather than from C++.

Reply:
Thanks for the pointer!! Added such a test for `aten::stack` in PR #74077.